projects/storefrontlib/src/shared/services/promotion/abstract-promotion.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Cart, Order, OrderEntry, PromotionResult } from '@spartacus/core';
```

Comment:
How about this:
```
@Injectable()
export abstract class PromotionService {
  protected loacatin: PromotionLocation;
  
  abstract getOrderPromotions();

  // then the functions used by children
  getProductPromotionForEntry;
  getProductPromotion;
  isConsumedByEntry;
```
Then in checkout:
```
@Injectable({
  providedIn: 'root',
})
export class CheckoutPromotionService extends PromotionService {
  location = PromotionLocation.Checkout;

  // then implement the getOrderPromotions(), also move the getOrderPromotionsFromOrderHelper to this class.
```
When use it in checkout:
```
providers: [
            {
              provide: PromotionService,
              useExisting: CheckoutPromotionService,
            },
          ],
```
same thing for the `Order` and `ActiveCart`

Reply:
`getOrderPromotion` maybe can be renamed to `getPromotion`?